conkit/core/contactmap.py

```diff
@@ -394,6 +394,29 @@ def singletons(self):
                 singletons.remove(contact.id)
         return singletons
 
+    @property
+    def highest_contact(self):
+        """The :obj:`~conkit.core.contact.Contact` with the highest residue sequence number in the :obj:`~conkit.core.contactmap.ContactMap`
+
+        Returns
+        -------
+        :obj:`~conkit.core.contact.Contact`
+           A :obj:`~conkit.core.contact.Contact` object
+
+        """
+        if len(self) == 0:
+            return None
+
+        highest_index = 0
+        highest_id = None
+
+        for contactid in self.as_list():
```

Comment:
not a good idea, you're iterating over all contacts to turn them into a list, then you're iterating over the list again. Despite being O(N), this makes no sense. either you use `for contact in self:` or even better you re-implement `Entity` to behave like a priority queue whereby sort-order can be res1, res2 or score